### ISSUE TITLE
S3906 false positive when implementing interface from 3rd party

### ISSUE THREAD

**Issue by nzain** [SonarSource/sonar-dotnet#3453]
### Description
SonarLint reports [S3906](https://rules.sonarsource.com/csharp/RSPEC-3906) violation (Event Handlers should have the correct signature) when I implement an interface from an external library. The defined delegates even look ok (according to the rule). Even if they were non-compliant, I couldn't change them because they are defined in the referenced 3rd party library.

### Repro steps
Create a new project and install [WPFLocalizeExtension](https://github.com/XAMLMarkupExtensions/WPFLocalizationExtension) via nuget.

Create a new class that implements the [`ILocalizationProvider.cs`](https://github.com/XAMLMarkupExtensions/WPFLocalizationExtension/blob/master/src/Providers/ILocalizationProvider.cs) interface. Three eventhandlers are defined via delegates in [`ProviderEventArgs.cs`](https://github.com/XAMLMarkupExtensions/WPFLocalizationExtension/blob/master/src/Providers/ProviderEventArgs.cs) and SonarLint will warn about S3906 rule violations. I have tried the following code:

```cs
    public class LocalizationProvider : ILocalizationProvider
    {
        public event ProviderChangedEventHandler ProviderChanged;

        public event ProviderErrorEventHandler ProviderError;
        
        public event ValueChangedEventHandler ValueChanged
        {
            add
            {
                // not implemented
            }
            remove
            {
                // not implemented
            }
        }

        // other interface members not shown here
    }
```

### Expected behavior

No rule violation should appear.

### Actual behavior

SonarLint complains about S3906, even if the code looks correct to me (and I couldn't change it anyways).

### Known workarounds

I couldn't find any workaround (except for ignoring another rule).

### Related information

* C#/VB.NET Plugins Version: SonarLint 4.23.0.19399 (July 7th 2020)
* Visual Studio Version: 2019 v16.6.4
* Operating System: Windows 10

**Comment by costin-zaharia-sonarsource** [comment]
Hi @nzain 

I can confirm the issue. Indeed we should not raise warnings when implementing interfaces or overriding inherited members.
